src/sync.ts

```diff
@@ -1350,13 +1362,15 @@ export const doActualSync = async (
     return;
   }
 
+  // If the metadata is synced first then we can't state if a sync was successful or not. Unless it is uploaded again after?
```

Comment:
We probably need to upload metadata after a sync, I think this makes sense to me.

Some failed syncs may result in metadata not being uploaded, which is acceptable (not all failures can be sent remotely, like if remote isn't responding or we don't have internet connection).

If a sync fails and we can't send that to the remote can we still show it as a "failed sync" in the local sync history?

Reply:
Thanks for reviewing. I think so as well.

I think having a history array and fetching metadata on plugin load could work to store local sync failures. But any failed syncs added locally wouldn't show after plugin reload. Regardless, I do need some way to access the history in historyView.ts so storing a public array in main.ts might help.

Reply:
Yeah can store this in the plugin settings for now which is persisted. I think this is worth doing  - there is also a local DB which I haven't explored much, could be a better place for persistence.